mlflow/models/evaluation/base.py

```diff
@@ -1695,29 +1713,14 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
                     "parameter when model=None.",
                     error_code=INVALID_PARAMETER_VALUE,
                 )
-            if predictions not in data.columns:
-                raise MlflowException(
-                    message=f"The specified predictions column '{predictions}' is not "
-                    "found in the specified data.",
-                    error_code=INVALID_PARAMETER_VALUE,
-                )
         elif isinstance(data, mlflow.data.pandas_dataset.PandasDataset):
             # If data is a mlflow PandasDataset with predictions specified
             # check that exact one predictions column is specified
-            if data.predictions is not None:
-                if predictions is not None and predictions != data.predictions:
-                    raise MlflowException(
-                        message="The predictions parameter must be None or the same as "
-                        "data.predictions when data.predictions is specified. Found "
-                        f"predictions='{predictions}', data.predictions='{data.predictions}'.",
-                        error_code=INVALID_PARAMETER_VALUE,
-                    )
-                else:  # predictions is None or predictions == data.predictions
-                    pass  # OK: exact one predictions column is specified
-            else:
```

Comment:
Simplifying the error handling case by not supporting `predictions == data.predictions`, please see the PR description for the supported case matrix.